indico/modules/events/abstracts/controllers/email_templates.py

```diff
@@ -108,15 +133,24 @@ class RHDeleteEmailTemplate(RHEditEmailTemplateBase):
     def _process(self):
         db.session.delete(self.email_tpl)
         tpl = get_template_module('events/abstracts/management/_notification_tpl_list.html')
-        return jsonify_data(html=tpl.render_notification_list(self.event_new), flash=False)
-
-
-class RHPreviewEmailTemplate(RHEditEmailTemplateBase):
-    """Preview an e-mail template."""
-
-    def _process(self):
-        self.commit = False
-        abstract = create_mock_abstract(self.event_new)
-        tpl = get_abstract_notification_tpl_module(self.email_tpl, abstract)
-        return jsonify_template('events/abstracts/management/notification_preview.html',
-                                subject=tpl.get_subject(), body=tpl.get_body())
+        track_dict, abstract_states, contr_type_dict, email_tpl_dict = _get_rules_fields(self.event_new)
+        return jsonify_data(html=tpl.render_notification_list(self.event_new,
+                                                              track=track_dict,
+                                                              states=abstract_states,
+                                                              contr_type=contr_type_dict,
+                                                              email_tpl=email_tpl_dict), flash=False)
+
+
+def _get_rules_fields(event_new):
+    """ Retrieve needed rules fields: Track, Abstract state and Contribution type. """
+    track_dict = {track.id: track for track in Track.query.with_parent(event_new)}
+    abstract_states = AbstractState.__titles__
+    contr_type_dict = {contr_type.id: contr_type
+                       for contr_type in ContributionType.query.with_parent(event_new)}
+    email_tpl_dict = {email_template.id: email_template
+                      for email_template in event_new.abstract_email_templates}
+    abstract = create_mock_abstract(event_new)
+    for k, v in email_tpl_dict.iteritems():
+        email_tpl_dict[k] = get_abstract_notification_tpl_module(v, abstract)
+
+    return track_dict, abstract_states, contr_type_dict, email_tpl_dict
```

Comment:
Why not returning a dictionary? It seems you build it only for it to be passed to `tpl.render_notification_list`. That way you could simply do `tpl.render_notification_list(**_get_rules_fields(self.event_new))`